src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBuffer.cs

```diff
@@ -1097,5 +1133,65 @@ private void ThrowIfNull()
                 throw new SqlNullValueException();
             }
         }
+
```

Comment:
This somment explains the jit behaviour and purpose of these strange looking methods.